tests/test_butler.py

```diff
@@ -205,17 +211,32 @@ def assertGetComponents(
             self.assertEqual(result_deferred, result)
 
     def tearDown(self) -> None:
-        removeTestTempDir(self.root)
+        if self.root is not None:
+            removeTestTempDir(self.root)
 
     def create_butler(
         self, run: str, storageClass: StorageClass | str, datasetTypeName: str
-    ) -> tuple[DirectButler, DatasetType]:
+    ) -> tuple[Butler, DatasetType]:
+        """Create a Butler for the test repository and insert some test data
+        into it.
+        """
         butler = Butler.from_config(self.tmpConfigFile, run=run)
```

Comment:
Can this call `self.create_empty_butler()` here? Or is there some restriction on this having to be a direct butler? If it can't call it, maybe add a comment to that effect?

Reply:
Yeah that makes sense and will let me get rid of the other method override below.  I didn't discover the need for `create_empty_butler` until later in the process so I didn't think about it.